client/src/pages/ProfileList/List.tsx

```diff
@@ -89,47 +107,31 @@ export default function List(): JSX.Element {
             </Box>
           </Grid>
         </Grid>
-        <Grid container justify="center" spacing={6} className={`${classes.root}`}>
-          {profiles.map((profile, key) => {
-            const { firstName, lastName, user, description } = profile;
-            return (
-              user && (
-                <Grid item key={user}>
-                  <Card style={{ height: '100%' }} variant="outlined" key={key}>
-                    <CardActionArea component={Link} to={`profile/${user}`}>
-                      <Grid
-                        justify="center"
-                        alignItems="center"
-                        direction="column"
-                        container
-                        className={`${classes.card}`}
-                      >
-                        <Avatar src={profilePhoto} className={classes.avatar} />
-
-                        <CardContent>
-                          <Typography className={classes.fullName} variant="h6">
-                            {firstName} {lastName}
-                          </Typography>
-                          <Rating name="read-only" value={placeholder.rating} readOnly />
+        <Grid container spacing={10} className={`${classes.root}`} sm>
+          {profiles
+            .filter((profile: Profile) => {
+              if (profile.address && profile.address.includes(searchTerm)) {
+                return profile;
+              }
+            })
+            .filter((profile: Profile) => {
+              //TODO: Implement date search
+              // const requestDays = moment.range(new Date(selection.startDate), new Date(selection.endDate));
 
-                          <Typography>{description ? description : defaultDescription}</Typography>
-                        </CardContent>
-                      </Grid>
-                    </CardActionArea>
-                    <Divider light />
-                    <Grid container className={`${classes.bottom}`} spacing={2}>
-                      <Grid item>
-                        <Typography className={classes.location}>{placeholder.location}</Typography>
-                      </Grid>
-                      <Grid item>
-                        <Typography className={classes.price}>${placeholder.price}/hr</Typography>
-                      </Grid>
-                    </Grid>
-                  </Card>
-                </Grid>
-              )
-            );
-          })}
+              // const days = Array.from(requestDays.by('day'));
+              // days.map((day) => {
+              //   if (day) {
+              //     const weekday = moment(day).format('dddd');
+              //     if (availability[weekday as key] === false) {
+              //       return;
+              //     }
+              //   }
+              // });
```

Comment:
As I understand, you want to keep this code for the future. In cases like this, it would be better to move this into a single-purpose function (e.g. `filterByDateRange(searchRange, availability)`) outside (or on top) of the file and leave a comment to give some context about the situation for other contributors to get on the same page 👍🏼

Reply:
After some thought, to use the implementation that you're proposing would this be how you'd do that? I was thinking there'd be a way to remove .filter as well and move it to the single purpose function? 
Let me know if this is what you are meaning 
`
            .filter((profile: Profile) => {
              locationfilter(profile)
              
            })
            .filter((profile: Profile) => {
            datefilter(profile)
            })

`

Reply:
I got it working! 

```          {profiles
            .filter((profile: Profile) => {
              return locationFilter(dateFilter(profile));
            })
            .map((profile, key) => (
              <UserCard profile={profile} key={key} />
            ))}
            
            ```

Reply:
This would work, yes. We'd have to return a value in filter though, idk something like:
```
    .filter((profile: Profile) => {
              return addressInRequestRange(location, profile.address);
            })
            .filter((profile: Profile) => {
              return availabilityCheck(requestDateRange, profile.availability);
            })
```